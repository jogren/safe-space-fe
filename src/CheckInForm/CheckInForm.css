.CheckInForm {
  display: flex;
  flex-direction: column;
  width: 50%;
  margin: auto;
  height: 80vh;
  border: 2px solid black;
  background-color: black;
  opacity: .7;
  margin-top: 40px;
  color: white;
  border-radius: 10px;
  font-weight: bold;
  overflow-y: scroll;
}

.CheckInForm-successful-checkin {
  display: flex;
  flex-direction: column;
  width: 30%;
  margin: auto;
  margin-top: 10%;
}

.CheckInForm-successful-checkin p {
  font-size: 20px;
  text-align: center;
}

.button_successful-checkin {
  height: 30px;
  margin-bottom: 10px;
  font-size: 14px;
  width: 100%;
}