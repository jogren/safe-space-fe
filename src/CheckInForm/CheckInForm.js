import React, { useState } from "react";
import BasicInfoForm from "../BasicInfoForm/BasicInfoForm";
import NeedsForm from "../NeedsForm/NeedsForm";
import EmergencyContactForm from "../EmergencyContactForm/EmergencyContactForm";
import { postNewUser, postNeeds, postEmergencyContacts } from "../apiCalls/apiCalls";
import "./CheckInForm.css";

const CheckInForm = ({ reliefCenter }) => {
  const [personName, setPersonName] = useState("");
  const [personAge, setPersonAge] = useState("");
  const [personPhone, setPersonPhone] = useState("");
  const [items, setItems] = useState([
    "diapers",
    "baby wipes",
    "breastfeeding supplies",
    "infant formula",
    "feminine products",
    "phone charger (iphone)",
    "phone charger (android)",
    "add item"
  ]);

  const [emergencyName, setEmergencyName] = useState("");
  const [emergencyPhone, setEmergencyPhone] = useState("");
  const [emergencyRelationship, setEmergencyRelationship] = useState("");
  const [sendMessage, setSendMessage] = useState(false)

  const submitUser = async e => {
    e.preventDefault();
    const personData = {
      name: personName,
      age: personAge,
      phone: personPhone,
      needs: items,
      emergencyName: emergencyName,
      emergencyPhone: emergencyPhone,
      emergencyRelationship: emergencyRelationship,
      notify: sendMessage
    };
    let userId = await postNewUser(personData, reliefCenter);

    await postNeeds(userId, [])
    await postEmergencyContacts(userId, personData)
  };

  return (
    <section className="CheckInForm">
      <BasicInfoForm
        personName={personName}
        setPersonName={setPersonName}
        personAge={personAge}
        setPersonAge={setPersonAge}
        personPhone={personPhone}
        setPersonPhone={setPersonPhone}
      />
      <NeedsForm items={items} setItems={setItems} />
      <EmergencyContactForm
        emergencyName={emergencyName}
        setEmergencyName={setEmergencyName}
        emergencyPhone={emergencyPhone}
        setEmergencyPhone={setEmergencyPhone}
        emergencyRelationship={emergencyRelationship}
        setEmergencyRelationship={setEmergencyRelationship}
        sendMessage={sendMessage}
        setSendMessage={setSendMessage}
      />
        <button id="submit" onClick={submitUser}>Submit</button>
    </section>
  );
};

export default CheckInForm;
