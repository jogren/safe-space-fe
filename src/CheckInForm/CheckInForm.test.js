import React from "react";
import { mount } from "enzyme";
import CheckInForm from "./CheckInForm";
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

describe("CheckInForm", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(
      <CheckInForm
      reliefCenter={{        
        name: "Turing Relief Center",
        addressPrint: "1331 17th St Denver CO 80202",
        lat: "39.7508132",
        lng: "-104.9967997",
        phone: "5555555555",
        website: "www.relief-center.com",
        id: "1"
      }}
      />
    );
  });

  it("should render without crashing", () => {
    expect(wrapper.length).toEqual(1);
  });

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});