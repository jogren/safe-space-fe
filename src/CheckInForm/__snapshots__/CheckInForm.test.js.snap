// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckInForm should match the snapshot 1`] = `
<CheckInForm
  reliefCenter={
    Object {
      "addressPrint": "1331 17th St Denver CO 80202",
      "id": "1",
      "lat": "39.7508132",
      "lng": "-104.9967997",
      "name": "Turing Relief Center",
      "phone": "5555555555",
      "website": "www.relief-center.com",
    }
  }
>
  <section
    className="CheckInForm"
  >
    <BasicInfoForm
      personAge=""
      personName=""
      personPhone=""
      setPersonAge={[Function]}
      setPersonName={[Function]}
      setPersonPhone={[Function]}
    >
      <form
        className="BasicInfoForm"
      >
        <h1
          id="basic-info-header"
        >
          Basic Information:
        </h1>
        <div
          id="fullname-container"
        >
          <label
            htmlFor="fullname"
            id="fullname-label"
          >
            Full Name:
          </label>
          <input
            id="fullname"
            name="name"
            onChange={[Function]}
            placeholder="Add Full Name"
            type="text"
            value=""
          />
        </div>
        <div
          id="age-container"
        >
          <label
            htmlFor="age"
            id="age-label"
          >
            Age:
          </label>
          <input
            id="age"
            name="age"
            onChange={[Function]}
            placeholder="Add Age"
            type="text"
            value=""
          />
        </div>
        <div
          id="phone-container"
        >
          <label
            htmlFor="phone"
            id="phone-label"
          >
            Phone Number:
          </label>
          <input
            id="phone"
            name="phone"
            onChange={[Function]}
            placeholder="Add Phone Number"
            type="text"
            value=""
          />
        </div>
      </form>
    </BasicInfoForm>
    <NeedsForm
      items={
        Array [
          "Diapers",
          "Baby Wipes",
          "Breastfeeding Supplies",
          "Infant Formula",
          "Feminine Products",
          "Phone Charger (iphone)",
          "Phone Charger (android)",
          "add item",
        ]
      }
      neededItems={Array []}
      setItems={[Function]}
      setNeededItems={[Function]}
    >
      <section
        className="NeedsForm"
      >
        <h1
          id="necessities-header"
        >
          Necessities:
        </h1>
        <article
          id="items-container"
        >
          <Item
            item="Diapers"
            items={
              Array [
                "Diapers",
                "Baby Wipes",
                "Breastfeeding Supplies",
                "Infant Formula",
                "Feminine Products",
                "Phone Charger (iphone)",
                "Phone Charger (android)",
                "add item",
              ]
            }
            key="0"
            neededItems={Array []}
            setItems={[Function]}
            setNeededItems={[Function]}
          >
            <article
              className="Item"
            >
              <div
                id="checkbox-item-container"
              >
                <img
                  alt=""
                  id="check-or-uncheck"
                  onClick={[Function]}
                  src={
                    Object {
                      "ReactComponent": Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "render": [Function],
                      },
                      "__esModule": true,
                      "default": "unchecked.svg",
                    }
                  }
                />
                <h3
                  id="item-name"
                  onClick={[Function]}
                >
                  Diapers
                </h3>
              </div>
            </article>
          </Item>
          <Item
            item="Baby Wipes"
            items={
              Array [
                "Diapers",
                "Baby Wipes",
                "Breastfeeding Supplies",
                "Infant Formula",
                "Feminine Products",
                "Phone Charger (iphone)",
                "Phone Charger (android)",
                "add item",
              ]
            }
            key="1"
            neededItems={Array []}
            setItems={[Function]}
            setNeededItems={[Function]}
          >
            <article
              className="Item"
            >
              <div
                id="checkbox-item-container"
              >
                <img
                  alt=""
                  id="check-or-uncheck"
                  onClick={[Function]}
                  src={
                    Object {
                      "ReactComponent": Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "render": [Function],
                      },
                      "__esModule": true,
                      "default": "unchecked.svg",
                    }
                  }
                />
                <h3
                  id="item-name"
                  onClick={[Function]}
                >
                  Baby Wipes
                </h3>
              </div>
            </article>
          </Item>
          <Item
            item="Breastfeeding Supplies"
            items={
              Array [
                "Diapers",
                "Baby Wipes",
                "Breastfeeding Supplies",
                "Infant Formula",
                "Feminine Products",
                "Phone Charger (iphone)",
                "Phone Charger (android)",
                "add item",
              ]
            }
            key="2"
            neededItems={Array []}
            setItems={[Function]}
            setNeededItems={[Function]}
          >
            <article
              className="Item"
            >
              <div
                id="checkbox-item-container"
              >
                <img
                  alt=""
                  id="check-or-uncheck"
                  onClick={[Function]}
                  src={
                    Object {
                      "ReactComponent": Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "render": [Function],
                      },
                      "__esModule": true,
                      "default": "unchecked.svg",
                    }
                  }
                />
                <h3
                  id="item-name"
                  onClick={[Function]}
                >
                  Breastfeeding Supplies
                </h3>
              </div>
            </article>
          </Item>
          <Item
            item="Infant Formula"
            items={
              Array [
                "Diapers",
                "Baby Wipes",
                "Breastfeeding Supplies",
                "Infant Formula",
                "Feminine Products",
                "Phone Charger (iphone)",
                "Phone Charger (android)",
                "add item",
              ]
            }
            key="3"
            neededItems={Array []}
            setItems={[Function]}
            setNeededItems={[Function]}
          >
            <article
              className="Item"
            >
              <div
                id="checkbox-item-container"
              >
                <img
                  alt=""
                  id="check-or-uncheck"
                  onClick={[Function]}
                  src={
                    Object {
                      "ReactComponent": Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "render": [Function],
                      },
                      "__esModule": true,
                      "default": "unchecked.svg",
                    }
                  }
                />
                <h3
                  id="item-name"
                  onClick={[Function]}
                >
                  Infant Formula
                </h3>
              </div>
            </article>
          </Item>
          <Item
            item="Feminine Products"
            items={
              Array [
                "Diapers",
                "Baby Wipes",
                "Breastfeeding Supplies",
                "Infant Formula",
                "Feminine Products",
                "Phone Charger (iphone)",
                "Phone Charger (android)",
                "add item",
              ]
            }
            key="4"
            neededItems={Array []}
            setItems={[Function]}
            setNeededItems={[Function]}
          >
            <article
              className="Item"
            >
              <div
                id="checkbox-item-container"
              >
                <img
                  alt=""
                  id="check-or-uncheck"
                  onClick={[Function]}
                  src={
                    Object {
                      "ReactComponent": Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "render": [Function],
                      },
                      "__esModule": true,
                      "default": "unchecked.svg",
                    }
                  }
                />
                <h3
                  id="item-name"
                  onClick={[Function]}
                >
                  Feminine Products
                </h3>
              </div>
            </article>
          </Item>
          <Item
            item="Phone Charger (iphone)"
            items={
              Array [
                "Diapers",
                "Baby Wipes",
                "Breastfeeding Supplies",
                "Infant Formula",
                "Feminine Products",
                "Phone Charger (iphone)",
                "Phone Charger (android)",
                "add item",
              ]
            }
            key="5"
            neededItems={Array []}
            setItems={[Function]}
            setNeededItems={[Function]}
          >
            <article
              className="Item"
            >
              <div
                id="checkbox-item-container"
              >
                <img
                  alt=""
                  id="check-or-uncheck"
                  onClick={[Function]}
                  src={
                    Object {
                      "ReactComponent": Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "render": [Function],
                      },
                      "__esModule": true,
                      "default": "unchecked.svg",
                    }
                  }
                />
                <h3
                  id="item-name"
                  onClick={[Function]}
                >
                  Phone Charger (iphone)
                </h3>
              </div>
            </article>
          </Item>
          <Item
            item="Phone Charger (android)"
            items={
              Array [
                "Diapers",
                "Baby Wipes",
                "Breastfeeding Supplies",
                "Infant Formula",
                "Feminine Products",
                "Phone Charger (iphone)",
                "Phone Charger (android)",
                "add item",
              ]
            }
            key="6"
            neededItems={Array []}
            setItems={[Function]}
            setNeededItems={[Function]}
          >
            <article
              className="Item"
            >
              <div
                id="checkbox-item-container"
              >
                <img
                  alt=""
                  id="check-or-uncheck"
                  onClick={[Function]}
                  src={
                    Object {
                      "ReactComponent": Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "render": [Function],
                      },
                      "__esModule": true,
                      "default": "unchecked.svg",
                    }
                  }
                />
                <h3
                  id="item-name"
                  onClick={[Function]}
                >
                  Phone Charger (android)
                </h3>
              </div>
            </article>
          </Item>
          <Item
            item="add item"
            items={
              Array [
                "Diapers",
                "Baby Wipes",
                "Breastfeeding Supplies",
                "Infant Formula",
                "Feminine Products",
                "Phone Charger (iphone)",
                "Phone Charger (android)",
                "add item",
              ]
            }
            key="7"
            neededItems={Array []}
            setItems={[Function]}
            setNeededItems={[Function]}
          >
            <article
              className="Item"
            >
              <div
                id="add-item-container"
              >
                <input
                  id="new-item"
                  name="name"
                  onChange={[Function]}
                  placeholder="Add Item/Medication"
                  type="text"
                  value=""
                />
                <button
                  id="submit-new-item"
                  onClick={[Function]}
                  type="submit"
                >
                  <img
                    alt="plus symbol"
                    id="plus-img"
                    src={
                      Object {
                        "ReactComponent": Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "render": [Function],
                        },
                        "__esModule": true,
                        "default": "plus-sign.svg",
                      }
                    }
                  />
                </button>
              </div>
            </article>
          </Item>
        </article>
      </section>
    </NeedsForm>
    <EmergencyContactForm
      emergencyName=""
      emergencyPhone=""
      emergencyRelationship=""
      sendMessage={false}
      setEmergencyName={[Function]}
      setEmergencyPhone={[Function]}
      setEmergencyRelationship={[Function]}
      setSendMessage={[Function]}
    >
      <form
        className="EmergencyContactForm"
      >
        <h1
          id="emergency-contact-info-header"
        >
          Emergency Contact Information:
        </h1>
        <div
          id="field"
        >
          <label
            htmlFor="Full Name"
            id="fullname-label"
          >
            Full Name:
          </label>
          <input
            id="fullname"
            name="name"
            onChange={[Function]}
            placeholder="Add Full Name"
            type="text"
            value=""
          />
        </div>
        <div
          id="field"
        >
          <label
            htmlFor="phone"
            id="phone-label"
          >
            Phone:
          </label>
          <input
            id="phone"
            name="phone"
            onChange={[Function]}
            placeholder="Add Phone Number"
            type="text"
            value=""
          />
        </div>
        <div
          id="field"
        >
          <label
            htmlFor="relationship"
            id="relationship-label"
          >
            Relationship:
          </label>
          <input
            id="relationship"
            name="relationship"
            onChange={[Function]}
            placeholder="Add Relationship"
            type="text"
            value=""
          />
        </div>
        <div
          id="make-public-or-private-container"
        >
          <h3
            id="public-or-private-button-text"
          >
            By clicking the following button, I agree to allow a message to be sent to the emergency contact number listed above.
          </h3>
          <button
            id="make-public-or-private-button"
            onClick={[Function]}
          >
            Send Text to Contact
          </button>
        </div>
      </form>
    </EmergencyContactForm>
    <hr />
    <div
      id="submit-form-button-container"
    >
      <h3
        id="submit-form-text"
      >
        By clicking the "Submit Form" button below, I certify that all information in this form is true and correct to the best of my knowledge.
      </h3>
      <button
        id="submit-form-button"
        onClick={[Function]}
      >
        Submit Form
      </button>
    </div>
  </section>
</CheckInForm>
`;
